File: Iteration 1/code/CSE3063GRP12/src/core/general_providers/SessionController.java
package core.general_providers;
import core.enums.UserType;
import core.models.abstracts.User;
import core.models.concretes.Advisor;
import core.models.concretes.Student;


public class SessionController {
    private User currentUser;
    private static SessionController instance;

    public static SessionController getInstance(){
        if (instance == null) {
            instance = new SessionController();
        }
        return instance;
    }
    private SessionController(){

    }
    public User getCurrentUser(){
        return currentUser;
    }

    public void setCurrentUser(User user){
        this.currentUser = user;
    }

    public UserType getUserType(){
        if(currentUser instanceof Student){
            return UserType.Student;
        }  
        else if(currentUser instanceof Advisor){
            return UserType.Advisor;
        }
        else{
            return null;
        }
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/general_providers/SessionController.py
from core.models.concretes.Advisor import Advisor
from core.models.concretes.Student import Student

class SessionController:
    _instance = None
    _current_user = None

    def __new__(cls):
        if not cls._instance:
            cls._instance = super(SessionController, cls).__new__(cls)
        return cls._instance

    @classmethod
    def getInstance(cls):
        if cls._instance is None:
            cls._instance = SessionController()
        return cls._instance

    def get_current_user(cls):
        return cls._current_user


    def set_current_user(cls, user):
        cls._current_user = user

    def get_user_type(cls):
        if isinstance(cls._current_user, Student):
            return "Student"
        elif isinstance(cls._current_user, Advisor):
            return "Advisor"
        else:
            return None